Sources/core/objects/Projectiles.cpp

```diff
@@ -0,0 +1,254 @@
+#include "Projectiles.h"
+#include "../Helpers.h"
+
+
+Projectile::Projectile(size_t id) : IMovable(id)
+{
+    SetPassable(D_ALL, Passable::AIR);
+    tickSpeed = 0.25;
+    damage = 0;
+    SetSprite("icons/projectiles.dmi");
+    v_level = 5;
+}
+void Projectile::Process()
+{
+    if(set_target_ == 0)
+    {
+       return;
+    }
+    int z = current_step_;
+    Dir step = VDirToDir(movement_[z]);
+
+   if(!TryMove(step))
```

Comment:
I think, you should not use `TryMove` here. `TryMove` function is used for some "right" movements (this function is called by force processor for example).
I suggest something like:
```
Dir new_dir = ... // somehow calculate next dir in the movement pattern
Rotate(new_dir); // rotate to new dir
if (!CheckPassable()) // if we cannot move - then projectile path has been ended
{
    Delete();
    return;
}
```
`CheckPassable` call inside itself `Bump` function on the object which does not allow pass.
So when projectile will hit human `Human::Bump(id_ptr_on<IMovable>)` will be called. And you should add some code into this `Bump` function in order to handle projectiles.